docs/tutorial/t-setup.md

```diff
@@ -0,0 +1,262 @@
+sudo microk8s enable metallb:$IPADDR-$IPADDR
+```
+
+Once done, the list of enabled addons can be seen via `microk8s status --wait-ready` command. The output of the command should look similar to the following:
+
+```
+microk8s is running
+...
+addons:
+  enabled:
+    dns                  # (core) CoreDNS
+    ha-cluster           # (core) Configure high availability on the current node
+    helm                 # (core) Helm - the package manager for Kubernetes
+    helm3                # (core) Helm 3 - the package manager for Kubernetes
+    hostpath-storage     # (core) Storage class; allocates storage from host directory
+    metallb              # (core) Loadbalancer for your Kubernetes cluster
+    storage              # (core) Alias to hostpath-storage add-on, deprecated
+...
+```
+
+### Setup MinIO
+
+Spark can be configured to use S3 for object storage. However for simplicity, instead of using AWS S3, we're going to use an S3 compliant object storage library `minio`, an add-on for which is shipped by default in `microk8s` installation. Using MinIO, we can have an S3 compliant bucket created locally which is more convinient than AWS S3 for experimentation purposes. Let's enable the `minio` addon for MicroK8s.
+
+```bash
+sudo microk8s enable minio
+```
+
+For Spark to be able to access the local S3 bucket, we'll need to authenticate to MinIO using an access key and a secret key. These credentials are generated and stored as Kubernetes secret when the `minio` add-on is enabled. Let's fetch these credentials and export them as environment variables in order to use them later.
+
+```bash
+export ACCESS_KEY=$(kubectl get secret -n minio-operator microk8s-user-1 -o jsonpath='{.data.CONSOLE_ACCESS_KEY}' | base64 -d)
+export SECRET_KEY=$(kubectl get secret -n minio-operator microk8s-user-1 -o jsonpath='{.data.CONSOLE_SECRET_KEY}' | base64 -d)
+export S3_ENDPOINT=$(kubectl get service minio -n minio-operator -o jsonpath='{.spec.clusterIP}')
+export S3_BUCKET="spark-tutorial"
+```
+
+The MinIO add-on also provides with a Web UI which can be used to interact with the local S3 object storage. For us to be able to open it in the browser, we will need the IP address and port at which the MinIO Web UI is exposed. Let's fetch the MinIO web interface URL as follows:
+
+```bash
+MINIO_UI_IP=$(kubectl get service microk8s-console -n minio-operator -o jsonpath='{.spec.clusterIP}')
+MINIO_UI_PORT=$(kubectl get service microk8s-console -n minio-operator -o jsonpath='{.spec.ports[0].port}')
+export MINIO_UI_URL=$MINIO_UI_IP:$MINIO_UI_PORT
+echo $MINIO_UI_URL
+```
+
+On running the command above, you should see a combination of IP address and port. Let's open the URL in a web browser. In the login page that appears, the username is the access key and the password is the secret key we fetched earlier. These credentials can now be viewed simply by `echo`ing the variables `ACCESS_KEY` and `SECRET_KEY`:
```

Comment:
A suggestion for structure changes that increase clarity. Github markdown doesn't support code blocks inside code blocks so I can't use the ` ```suggestion` block for this one:

---

The MinIO add-on also provides with a Web UI which can be used to interact with the local S3 object storage. For us to be able to open it in the browser, we will need the IP address and port at which the MinIO Web UI is exposed. 
	
Let's fetch the MinIO web interface URL as follows:
```bash
MINIO_UI_IP=$(kubectl get service microk8s-console -n minio-operator -o jsonpath='{.spec.clusterIP}')
MINIO_UI_PORT=$(kubectl get service microk8s-console -n minio-operator -o jsonpath='{.spec.ports[0].port}')
export MINIO_UI_URL=$MINIO_UI_IP:$MINIO_UI_PORT
```
Your URL is a combination of an IP address and port. Print it by running 
```bash
echo $MINIO_UI_URL
```
Let's open this URL in a web browser. In the login page, the username is the access key and the password is the secret key we fetched earlier.
	
These credentials can now be viewed simply by `echo`ing the variables `ACCESS_KEY` and `SECRET_KEY`: